internal/catalogmetadata/types.go

```diff
@@ -0,0 +1,21 @@
+package catalogmetadata
+
+import (
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+)
+
+type Schemas interface {
+	Package | Bundle | Channel
+}
+
+type Package struct {
+	declcfg.Package
+}
+
+type Channel struct {
+	declcfg.Channel
+}
+
+type Bundle struct {
+	declcfg.Bundle
+}
```

Comment:
Thinking about how we can re-implement all the [predicates](https://github.com/operator-framework/operator-controller/blob/edd5ad9b195e56455ad47d292612494930027e0d/internal/resolution/util/predicates/predicates.go) from entities to this struct.

These should be easy:

* `WithPackageName`
* `InMastermindsSemverRange` and `InBlangSemverRange`
* `ProvidesGVK`
* `WithBundleImage`

As all the data we need is in `declcfg.Bundle`.

However `InChannel` and `Replaces` are tricky as `declcfg.Bundle` doesn't include channel membership info required for both.

I'm currently thinking about adding a file like `InChannels []*Channel` to `Bundle` struct which we will be populating when reading catalog metadata.

This should enable us relatively easily answer whether a bundle is in a specific channel or not.

`Replaces` is a bit tricker: we would need to go read each channel from the bundle and make sure that there is (or is not) a `ChannelEntry` which contains a replace with a specific value.